### ISSUE TITLE
Inconsistent validation of keywords as identifiers in method declaration in ES8

### ISSUE THREAD

**Issue by nikarh** [eslint/eslint#9153]
**Tell us about your environment**

* **ESLint Version:**
4.5.0

**Please show your full configuration:**
Reproducible in online REPL using a default configuration with ES8 enabled
[https://eslint.org/demo/](https://eslint.org/demo/)

**What did you do? Please include the actual source code causing the issue.**

```js
class SyncClass {
	delete() {}
}

const syncInstance = {
	delete() {}
}

class AsyncClass {
	async delete() {}
}

const asyncInstance = {
	async delete() {}
}
```

**What did you expect to happen?**
All "delete" method declarations should be either valid or invalid

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint fails to parse method declaration for `asyncInstance`.
```
14:8 - Parsing error: Unexpected token delete
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. This looks like a bug in the [acorn](https://github.com/ternjs/acorn) parser -- I've reported it at https://github.com/ternjs/acorn/issues/584.

**Comment by not-an-aardvark** [comment]
This has been fixed upstream in acorn, so it should be fixed in ESLint next time acorn is released.

**Comment by platinumazure** [comment]
Maybe we should leave this open and do an espree release and Upgrade commit against this issue when Acorn is released? Just to avoid any oddities for our users when it is available due to open dependency ranges?

**Comment by not-an-aardvark** [comment]
Sure, works for me.
